src/realm/db.cpp

```diff
@@ -2403,11 +2406,33 @@ bool DB::call_with_lock(const std::string& realm_path, CallbackWithLock callback
     return false;
 }
 
-std::vector<std::pair<std::string, bool>> DB::get_core_files(const std::string& realm_path)
+std::vector<std::pair<std::string, bool>> DB::get_core_files(const std::string& realm_path, uint64_t core_file_type)
 {
     std::vector<std::pair<std::string, bool>> files;
-    files.emplace_back(std::make_pair(realm_path, false));
-    files.emplace_back(std::make_pair(realm_path + ".management", true));
+    if ((DB::CoreFileType::Lock & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".lock", false));
+    }
+    if ((DB::CoreFileType::Storage & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path, false));
+    }
+    if ((DB::CoreFileType::Management & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".management", true));
+    }
+    if ((DB::CoreFileType::Note & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".note", false));
+    }
+    if ((DB::CoreFileType::Log & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".log", false));
+    }
+    if ((DB::CoreFileType::LogA & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".log_a", false));
+    }
+    if ((DB::CoreFileType::LogB & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".log_b", false));
+    }
+    if ((DB::CoreFileType::Backup & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".realm.backup", false));
+    }
```

Comment:
What do you think about throwing an exception `if (core_file_type > DB::CoreFileType::All)` ? That would catch people passing in unhandled inputs. But we could also choose to silently ignore unknown values. In either case, we should document the behaviour of unknown values.

Reply:
Addressed with 85e5bf504. That's basically obsolete now. :)